api/api/routers/mcp/mcp_server.py

```diff
@@ -693,5 +693,131 @@ async def create_api_key() -> LegacyMCPToolReturn:
     )
 
 
+@_mcp.tool()
+async def search_documentation(
+    query: str | None = Field(
+        default=None,
+        description="Search across all WorkflowAI documentation. Use this when you need to find specific information across multiple pages.",
+    ),
+    page: str | None = Field(
+        default=None,
+        description="Direct access to specific documentation page. Available pages are dynamically determined from the docsv2 directory structure.",
+    ),
+) -> LegacyMCPToolReturn:
+    """Search WorkflowAI documentation OR fetch a specific documentation page.
```

Comment:
Do we have a limit on how large the tool doc should be ? Maybe we could only return broad sections here and allow the tool to navigate the doc ?

Reply:
Add TODO for auto-generating page sections ?